presto-main/src/main/java/com/facebook/presto/spiller/FileSingleStreamSpiller.java

```diff
@@ -83,7 +82,19 @@ public FileSingleStreamSpiller(
         this.executor = requireNonNull(executor, "executor is null");
         this.spillerStats = requireNonNull(spillerStats, "spillerStats is null");
         this.localSpillContext = spillContext.newLocalSpillContext();
-        this.memoryContext = requireNonNull(memoryContext, "memoryContext can not be null");
+        this.memoryContext = requireNonNull(memoryContext, "memoryContext is null");
+        // HACK!
+        // The writePages() method is called in a separate thread pool and it's possible that
+        // these spiller threads can race with the driver thread that calls FileSingleStreamSpiller::close()
+        // in Driver::destroyIfNecessary().
+        // Due to this race when the spiller threads are running, the driver thread:
+        // 1. Can zero out the memory reservation even though the spiller threads physically hold onto that memory.
+        // 2. Can close/delete the temp file(s) used for spilling, which doesn't have any visible side effects, but still not desirable.
```

Comment:
Why this is not desirable?

I think if we are in a middle of query abort, then we could definitely close the spill file in order for the spiller to finish prematurely. The future would correctly hold an exception and it would be responsibility of caller to be aware that we are aborting.

It seems that currently there is an issue that the spiller will be running even if the query aborts. If the pages are large (e.g: tens of GBs) then such aborted spilling would take considerable time and resources.

Reply:
> If the pages are large
If the total size of pages is large...

Reply:
@nezihyigitbasi Please try to understand concern from @sopel39. See if the comments needs improvement. See if we misunderstood this class.

Reply:
I guess when we discussed this issue, we thought it was `not desirable`, because there is a race here -- file can be deleted while a spiller thread is running, other than that I don't think it causes any issues. If you have any proposals to clarify the javadoc further please let us know.